backend/app/Services/VisitorsFlow/FlowAggregateService.php

```diff
@@ -0,0 +1,409 @@
+    private $visitRepository;
+    private $visitorFlowCountryRepository;
+    private $websiteRepository;
+    private $geoPositionRepository;
+    private $visitorFlowBrowserRepository;
+    private $visitorFlowDeviceRepository;
+    private $visitorFlowScreenRepository;
+
+    public function __construct(
+        PageRepository $pageRepository,
+        VisitRepository $visitRepository,
+        VisitorFlowCountryRepository $visitorFlowCountryRepository,
+        WebsiteRepository $websiteRepository,
+        GeoPositionRepository $geoPositionRepository,
+        VisitorFlowBrowserRepository $visitorFlowBrowserRepository,
+        VisitorFlowDeviceRepository $visitorFlowDeviceRepository,
+        VisitorFlowScreenRepository $visitorFlowScreenRepository
+    ) {
+        $this->pageRepository = $pageRepository;
+        $this->visitRepository = $visitRepository;
+        $this->visitorFlowCountryRepository = $visitorFlowCountryRepository;
+        $this->websiteRepository = $websiteRepository;
+        $this->geoPositionRepository = $geoPositionRepository;
+        $this->visitorFlowBrowserRepository = $visitorFlowBrowserRepository;
+        $this->visitorFlowDeviceRepository = $visitorFlowDeviceRepository;
+        $this->visitorFlowScreenRepository = $visitorFlowScreenRepository;
+    }
+
+    public function aggregate(Visit $visit): void
+    {
+        $previousVisit = $this->getLastVisit($visit);
+        $isFirstInSession = $previousVisit === null;
+        $level = $this->getLevel($visit, $isFirstInSession);
+
+        $countryAggregate = $this->getCountryAggregate($visit, $level, $isFirstInSession, $previousVisit);
+        $browserAggregate = $this->getBrowserAggregate($visit, $level, $isFirstInSession, $previousVisit);
+        $deviceAggregate = $this->getDeviceAggregate($visit, $level, $isFirstInSession, $previousVisit);
+        $screenAggregate = $this->getScreenAggregate($visit, $level, $isFirstInSession, $previousVisit);
+
+        $this->updateCountryAggregate($visit, $level, $previousVisit, $countryAggregate);
+        $this->updateBrowserAggregate($visit, $level, $previousVisit, $browserAggregate);
+        $this->updateDeviceAggregate($visit, $level, $previousVisit, $deviceAggregate);
+        $this->updateScreenAggregate($visit, $level, $previousVisit, $screenAggregate);
+    }
+
+    private function getLevel(Visit $visit, bool $isFirstInSession): int
+    {
+        if (!$isFirstInSession) {
+            return $this->getVisitsCount($visit);
+        }
+        return 1;
+    }
+
+    private function updateCountryAggregate(
+        Visit $visit,
+        int $level,
+        Visit $previousVisit,
+        ?CountryAggregate $countryAggregate): void
+    {
+        if (!$countryAggregate) {
+            $countryAggregate = $this->createCountryAggregate($visit, $level, $previousVisit);
+            $this->visitorFlowCountryRepository->save($countryAggregate);
+            return;
+        }
+        if ($level > 1) {
```

Comment:
there is lots of duplicated methods, can you make it better

Reply:
if you have such statements in your method  "if ($level > 1) {" it means this method is responsible for more than one thing